### DISCUSSION TITLE
Filtering in server components

### DISCUSSION THREAD

**Discussion by teddybee** [vercel/next.js#52752] [App Router]
I would like to search/filter a list that is fetched in an async server component (page). The filter parameters are stored in client side (state manager).
Is there any workaround to do the filtering on the fetched list?
I wouldn`t put these busines logic to the plane simple <List /> component and also don't want to change the page into client component.

**Comment by masterkain** [comment]
maybe we can take a look at some code:

https://github.com/icoretech/airbroke/blob/main/components/Search.tsx
https://github.com/icoretech/airbroke/blob/main/lib/generateUpdatedUrl.ts

you would then parse the params for search/filtering:

https://github.com/icoretech/airbroke/blob/main/components/NoticesTable.tsx
https://github.com/icoretech/airbroke/blob/main/lib/queries/notices.ts#L22

basically you play with urls and params building/parsing, eventually (beside the search interaction) you can build your filters with simple links and keep rsc
hope this helps

**Comment by eideard-hm** [comment]
Just what I needed, it worked for me

Thank you very much,
Regards
